tutorials/polkadot-sdk/build-a-blockchain/add-trusted-nodes.md

```diff
@@ -0,0 +1,458 @@
+
+In blockchain networks, nodes must agree on the data state at any given time - a concept known as consensus. The [Polkadot-SDK Solochain Template](https://github.com/paritytech/polkadot-sdk-solochain-template){target=\_blank} uses a proof of authority consensus model called Aura (Authority Round), which limits block production to a rotating list of authorized accounts. These trusted participants, or authorities, create blocks in a round-robin fashion.
+
+This approach provides a simple method for launching a solo blockchain with a limited number of participants. 
+
+You'll learn how to generate keys for network authorities, create a custom chain specification file, and launch a private two-node blockchain network.
+
+## Prerequisites
+
+Before starting this tutorial, ensure you have:
+
+- Installed and configured Rust on your system. For detailed instructions on installing Rust and setting up your development environment, refer to the [Installation]() guide
+- Completed the [Build a Local Blockchain]() tutorial and have the [Polkadot-SDK Solochain Template](https://github.com/paritytech/polkadot-sdk-solochain-template){target=\_blank} installed on your local machine
+- Experience using predefined accounts to start nodes on a single computer, as described in the [Simulate a Network]() guide
+
+## Generate an Account and Keys
+
+Unlike in the [Simulate a Network]() tutorial where you used predefined accounts and keys to start peer nodes, this tutorial requires you to generate unique secret keys for your validator nodes. It's crucial to understand that in a real blockchain network, each participant is responsible for generating and managing their own unique set of keys.
+This process of generating your own keys serves several important purposes:
+
+- It enhances the security of your network by ensuring that each node has its own unique cryptographic identity
+- It simulates a more realistic blockchain environment where participants don't share key information
+- It helps you understand the process of key generation, which is a fundamental skill in blockchain operations
+
+### Key Generation Options
+
+### Generate Local Keys with the Node Template
+
+Best practices for key generation:
+
+- Ideally, use an air-gapped computer (never connected to the internet) when generating keys for a production blockchain
+- At minimum, disconnect from the internet before generating keys for any public or private blockchain not under your control
+
+For this tutorial, however, we'll use the `solochain-template-node `command-line options to generate random keys locally while remaining connected to the internet. This method is suitable for learning and testing purposes.
+
+Follow these steps to generate your keys:
+
+1. Navigate to the root directory where you compiled the node template
+
+2. Generate a random secret phrase and Sr25519 keys. Enter a password when prompted:
```

Comment:
Hey, @nhussein11! Do you think it needs more than this from the preceding section?

"To understand more about the different signing algorithms used in this tutorial (sr25519 and ed25519), check [Keypairs and Signing](https://wiki.polkadot.network/docs/learn-cryptography#keypairs-and-signing){target=\_blank}. "

If so, can you tell me who is working on updating this piece?

Reply:
@dawnkelly09 mm I think that should be enough to contextualize a just a bit...

Maybe link "Keypairs and Signing" with this reference: https://wiki.polkadot.network/docs/learn-cryptography#keypairs-and-signing 

> If so, can you tell me who is working on updating this piece?

I think @0xLucca since he is the author of this PR. @0xLucca if you want me to take this one just let me know :)

Reply:
Thats already included in the Generate an account and keys section:

`To understand more about the different signing algorithms used in this tutorial (sr25519 and ed25519), check [Keypairs and Signing](https://wiki.polkadot.network/docs/learn-cryptography#keypairs-and-signing). To learn more about the different types of keys used, refer to the [Keys](https://wiki.polkadot.network/docs/learn-cryptography#keys) section in the Polkadot Wiki.`